anesthetic/samples.py

```diff
@@ -281,6 +281,12 @@ def plot_2d(self, axes, *args, **kwargs):
                     lkwargs = local_kwargs.get(pos, {})
                     self.plot(ax_, x, y, plot_type=plot_type, *args, **lkwargs)
 
+        for y, row in axes.iterrows():
+            for x, ax in row.iteritems():
+                if x == y and ax is not None:
+                    axes[x][y] = axes[x][y].twin
+                    axes[x][y].twin = axes[x][y]
+
```

Comment:
This doesn't swap the axes, if that's what you're trying to do. The pythonic standard would be something like 
```python
a, b = b, a
```

Reply:
It might also to do this in the ``make_2d_axes`` code, rather than in the samples code, and to do it at the point of construction. Line 279 in this file could then be removed.

Reply:
> This doesn't swap the axes, if that's what you're trying to do.

I wasn't. Do we need access to the empty axis?

> do this in the `make_2d_axes` code

Briefly tried that, but something didn't work. Wasn't very persistent, though.

Reply:
> I wasn't. Do we need access to the empty axis?

I'm not sure we do. I don't think we need it to be user-accessible. I can think of use-cases where we use it to ensure that y-axes limits in the top-row are synced with the x-axes limit (see #25), but in that case, we could restore this functionality under a 'dangerous' underscored variable. 

If we're dropping twinning completely, the changes should definitely be in the ``make_2d_axes`` part.

Reply:
Hmm, labels and ticks are handled in the untwinned axes, though... We do want user access to them, right?

Reply:
Changed the behaviour in [9f00b05](https://github.com/williamjameshandley/anesthetic/pull/30/commits/9f00b05ffdb58b46c4b032a07fa396dde0205d60). Keep the twins the way they are, but pass on lines and patches from one to the other.

Reply:
OK. I think this is a nice and consistent solution. This is the reason that I had the 'twin' approach, in that it makes sense for the 1D axis to be distinct from, and plotted over the 2D axis that forms part of the grid. Passing the lines and patches is a nice solution, assuming it works. Is there anything else that should be passed over?

Reply:
The complete set of available handles would be {ax.lines, ax.patches, ax.collections, ax.containers}, but with `plot` and `hist` we only get the first two, so I only passed those. Would you prefer passing the full list?

Reply:
Unless it causes issues, it's probably best to pass them all. There may be extensions/use case that we haven't thought of which require the other two.

Reply:
done

Reply:
Ha, maybe...

Reply:
yes, seems to work, very nice